### PR TITLE
[TASK] Added description on how to add custom smarty plugins to theme…

### PR DESCRIPTION
**Pull request by wolv-dev**
… or plugin

### CODE REVIEW

**Comment by mitelg** [comment]
hey @wolv-dev 
thanks for your PR :+1: 
there is already a small section about custom smarty plugins: https://developers.shopware.com/designers-guide/smarty/#register-custom-smarty-plugins maybe you can adjust this section with your changes?

**Comment by wolv-dev** [comment]
@mitelg I moved the documentation and added reference

File: source/designers-guide/smarty-plugins/index.md [hunk 1]
```diff
@@ -216,3 +216,7 @@ The plugin uses widgets to embed templates. Widgets are self-contained parts of
 In this case, an HTTP request is triggered internally within the system, resulting in widgets being completely dynamic elements which are not cached.
 
 For further information read this blog: <a href="{{ site.url }}/blog/2016/07/11/on-action-tags/">On action tags</a>
+
+
+## Custom Smarty plugins
+To register custom smarty plugins please see [register custom smarty plugins](https://developers.shopware.com/designers-guide/smarty/#register-custom-smarty-plugins)
```
Comment (mitelg) [inline source/designers-guide/smarty-plugins/index.md]: you could use `{{ site.url }}/designers-guide/....` here

Comment (mitelg) [inline source/designers-guide/smarty-plugins/index.md]: you could use `{{ site.url }}/designers-guide/....` here

File: source/designers-guide/smarty/index.md [hunk 2]
```diff
@@ -199,6 +199,35 @@ In Shopware 5 we added the ability to register your own custom Smarty plugins in
...
 
 ```
-/_private/smarty/function.markdown.php
-/_private/smarty/modifier.picture.php
+themes/Frontend/ThemeName/_private/smarty/function.markdown.php
+themes/Frontend/ThemeName/_private/smarty/modifier.picture.php
+```
+**Notice:** Smarty plugins are not working inside mail templates
+
+#### Register custom smarty plugins in shopware plugin ####
```
Comment (mitelg) [inline source/designers-guide/smarty/index.md]:
```suggestion
### Register custom smarty plugins in a Shopware plugin
```

File: source/designers-guide/smarty/index.md [hunk 3]
```diff
@@ -199,6 +199,35 @@ In Shopware 5 we added the ability to register your own custom Smarty plugins in
...
+```php
+<?php
+namespace PluginName\Components\CompilerPass;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+/**
+ * Class AddTemplatePluginDirCompilerPass
+ */
+class AddTemplatePluginDirCompilerPass implements CompilerPassInterface
```
Comment (mitelg) [inline source/designers-guide/smarty/index.md]: is this necessary? afaik, there is also a convention how the directory has to look like 

@shyim to the rescue?

File: source/designers-guide/smarty-plugins/index.md (hunk 1 above)
Comment (wolv-dev) [reply source/designers-guide/smarty-plugins/index.md]: `{{ site.url }}` is only possible in a-tag. But I changed it.

File: source/designers-guide/smarty/index.md (hunk 2 above)
Comment (wolv-dev) [reply source/designers-guide/smarty/index.md]: Changed

File: source/designers-guide/smarty/index.md [hunk 4]
```diff
@@ -199,6 +199,35 @@ In Shopware 5 we added the ability to register your own custom Smarty plugins in
...
-/_private/smarty/function.markdown.php
-/_private/smarty/modifier.picture.php
+themes/Frontend/ThemeName/_private/smarty/function.markdown.php
+themes/Frontend/ThemeName/_private/smarty/modifier.picture.php
+```
+**Notice:** Smarty plugins are not working inside mail templates
+
+### Register custom smarty plugins in a Shopware plugin
+
+To specify your custom plugins directory inside your shopware plugin you can use the `CompilerPass` to add your own directory:
```
Comment (mitelg) [inline source/designers-guide/smarty/index.md]:
```suggestion
To specify your custom smarty plugin directory inside your Shopware plugin you can use the `CompilerPass` to add your own directory:
```

File: source/designers-guide/smarty/index.md [hunk 5]
```diff
@@ -199,6 +199,35 @@ In Shopware 5 we added the ability to register your own custom Smarty plugins in
...
+     * You can modify the container here before it is dumped to PHP code.
+     *
+     * @param ContainerBuilder $container
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $template = $container->getDefinition('template');
+        $template->addMethodCall('addPluginsDir', ['directory/name']);
+    }
+}
```
Comment (mitelg) [inline source/designers-guide/smarty/index.md]: I would also add a short code snippet, how the compiler pass is added to Shopware

**Comment by mitelg** [comment]
thanks for your contribution @wolv-dev :+1: :tada: :blue_heart:

**Pull request merged**